crypto/bio/bss_dgram.c

```diff
@@ -1147,18 +1147,17 @@ static int pack_local(BIO *b, MSGHDR_TYPE *mh, const BIO_ADDR *local) {
         return 1;
 
 #  elif defined(IP_SENDSRCADDR)
-        {
-            struct cmsghdr *cmsg;
-            struct in_addr *info;
+        struct cmsghdr *cmsg;
+        struct in_addr *info;
+        bio_dgram_data *data = b->ptr;
```

Comment:
Why not hoist this and the other declaration on line 1113 to line 1108?

Reply:
... and the same declarations from line 1176.  Both `cmsg` and `data` can be made common

Reply:
Because it would be unused in the else cases?

Reply:
Not if `IPV6_PKTINFO` is defined.

*The point is that these two variables are (or should be) declared exactly the same in three sections of this function*.  That's good enough reason to make it one and the same at the top...  and since the fault you corrected in this spot can be said to be caused by `data` lacking even though is was declared in the other section, that is also further justification to have it (along with `cmsg` common for all.

Reply:
But if none of these are defined then we would get a warning. But yeah, I can add another #if defined block.